// @thumbtack/thumbprint-tokens/dist/javascript-cjs/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.tpBorderRadiusBase = "4px";
exports.tpBorderRadiusBig = "6px";
exports.tpBorderRadiusFull = "50%";
exports.tpBorderRadiusSides = "9999px";
exports.tpBreakpointLarge = "1025px";
exports.tpBreakpointLargeValue = 1025;
exports.tpBreakpointMedium = "700px";
exports.tpBreakpointMediumValue = 700;
exports.tpBreakpointSmall = "481px";
exports.tpBreakpointSmallValue = 481;
exports.tpBreakpointSplitViewMedium = "1060px";
exports.tpBreakpointSplitViewMediumValue = 1060;
exports.tpBreakpointSplitViewSmall = "769px";
exports.tpBreakpointSplitViewSmallValue = 769;
exports.tpColorBlue100 = "#eaf6fa";
exports.tpColorBlue200 = "#b3ebff";
exports.tpColorBlue300 = "#79d2f2";
exports.tpColorBlue = "#009fd9";
exports.tpColorBlue500 = "#007fad";
exports.tpColorBlue600 = "#005a7a";
exports.tpColorIndigo100 = "#e8f1fd";
exports.tpColorIndigo200 = "#cce1ff";
exports.tpColorIndigo300 = "#96c2ff";
exports.tpColorIndigo = "#5968e2";
exports.tpColorIndigo500 = "#4f54b3";
exports.tpColorIndigo600 = "#383c80";
exports.tpColorPurple100 = "#f5efff";
exports.tpColorPurple200 = "#dfccff";
exports.tpColorPurple300 = "#d3baff";
exports.tpColorPurple = "#a97ff0";
exports.tpColorPurple500 = "#8a5ed6";
exports.tpColorPurple600 = "#502095";
exports.tpColorGreen100 = "#e1fdf3";
exports.tpColorGreen200 = "#c6f7da";
exports.tpColorGreen300 = "#73e4a2";
exports.tpColorGreen = "#2db783";
exports.tpColorGreen500 = "#109e78";
exports.tpColorGreen600 = "#0f6648";
exports.tpColorYellow100 = "#fdf7e7";
exports.tpColorYellow200 = "#ffebb3";
exports.tpColorYellow300 = "#ffdd80";
exports.tpColorYellow = "#febe14";
exports.tpColorYellow500 = "#d48300";
exports.tpColorYellow600 = "#8a5500";
exports.tpColorRed100 = "#ffeff0";
exports.tpColorRed200 = "#ffd9d9";
exports.tpColorRed300 = "#ffb0b0";
exports.tpColorRed = "#ff5a5f";
exports.tpColorRed500 = "#cc3553";
exports.tpColorRed600 = "#76012c";
exports.tpColorBlack300 = "#676d73";
exports.tpColorBlack = "#2f3033";
exports.tpColorGray200 = "#fafafa";
exports.tpColorGray300 = "#e9eced";
exports.tpColorGray = "#d3d4d5";
exports.tpColorWhite = "#ffffff";
exports.tpFontFamilyBase = "Mark, Avenir, Helvetica, Arial, sans-serif";
exports.tpFontFamilyMonospace = "'Source Code Pro', monospace";
exports.tpFontWeightNormal = "400";
exports.tpFontWeightBold = "700";
exports.tpFontTitle1Size = "28px";
exports.tpFontTitle1LineHeight = "32px";
exports.tpFontTitle1Weight = "700";
exports.tpFontTitle1ResponsiveSize = "40px";
exports.tpFontTitle1ResponsiveLineHeight = "44px";
exports.tpFontTitle1ResponsiveWeight = "700";
exports.tpFontTitle2Size = "24px";
exports.tpFontTitle2LineHeight = "28px";
exports.tpFontTitle2Weight = "700";
exports.tpFontTitle2ResponsiveSize = "32px";
exports.tpFontTitle2ResponsiveLineHeight = "40px";
exports.tpFontTitle2ResponsiveWeight = "700";
exports.tpFontTitle3Size = "22px";
exports.tpFontTitle3LineHeight = "28px";
exports.tpFontTitle3Weight = "700";
exports.tpFontTitle3ResponsiveSize = "24px";
exports.tpFontTitle3ResponsiveLineHeight = "32px";
exports.tpFontTitle3ResponsiveWeight = "700";
exports.tpFontTitle4Size = "20px";
exports.tpFontTitle4LineHeight = "28px";
exports.tpFontTitle4Weight = "700";
exports.tpFontTitle5Size = "18px";
exports.tpFontTitle5LineHeight = "24px";
exports.tpFontTitle5Weight = "700";
exports.tpFontTitle6Size = "16px";
exports.tpFontTitle6LineHeight = "24px";
exports.tpFontTitle6Weight = "700";
exports.tpFontTitle7Size = "14px";
exports.tpFontTitle7LineHeight = "20px";
exports.tpFontTitle7Weight = "700";
exports.tpFontTitle8Size = "12px";
exports.tpFontTitle8LineHeight = "18px";
exports.tpFontTitle8Weight = "700";
exports.tpFontBody1Size = "16px";
exports.tpFontBody1LineHeight = "24px";
exports.tpFontBody2Size = "14px";
exports.tpFontBody2LineHeight = "20px";
exports.tpFontBody3Size = "12px";
exports.tpFontBody3LineHeight = "18px";
exports.tpLetterSpacingLoose = "1px";
exports.tpLetterSpacingTight = "-1px";
exports.tpLetterSpacingExtraTight = "-2px";
exports.tpLineHeightBase = "1.6";
exports.tpLineHeightTight = "1.4";
exports.tpLineHeightLoose = "1.9";
exports.tpShadow100 = "0px 1px 3px rgba(0, 0, 0, 0.1)";
exports.tpShadow200 = "0px 2px 4px rgba(0, 0, 0, 0.15)";
exports.tpShadow300 = "0px 2px 7px rgba(0, 0, 0, 0.15)";
exports.tpShadow400 = "0px 2px 10px rgba(0, 0, 0, 0.2)";
// @deprecated
exports.tpShadowCard = "0 -1px 1px 0 rgba(0, 0, 0, 0.04), 0 1px 2px 0 rgba(0, 0, 0, 0.16)";
exports.tpSpace1 = "4px";
exports.tpSpace2 = "8px";
exports.tpSpace3 = "16px";
exports.tpSpace4 = "24px";
exports.tpSpace5 = "32px";
exports.tpSpace6 = "64px";
exports.tpSpace7 = "128px";
exports.tpSpace8 = "256px";
exports.tpWrapMaxWidth = "946px";
exports.tpWrapNoPadWidth = "1010px";
exports.tpZIndexModal = 200;
